amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/SynchronousBlockingRetrievalFactory.java

```diff
@@ -66,8 +84,18 @@ public GetRecordsRetrievalStrategy createGetRecordsRetrievalStrategy(@NonNull fi
         final StreamIdentifier streamIdentifier = shardInfo.streamIdentifierSerOpt().isPresent() ?
                 StreamIdentifier.multiStreamInstance(shardInfo.streamIdentifierSerOpt().get()) :
                 StreamIdentifier.singleStreamInstance(streamName);
-        return new SynchronousGetRecordsRetrievalStrategy(
-                new KinesisDataFetcher(kinesisClient, streamIdentifier, shardInfo.shardId(), maxRecords, metricsFactory, kinesisRequestTimeout));
+
+        final DataFetcherProviderConfig kinesisDataFetcherProviderConfig = new KinesisDataFetcherProviderConfig(streamIdentifier,
+                shardInfo.shardId(),
+                metricsFactory,
+                maxRecords,
+                kinesisRequestTimeout);
```

Comment:
nit: possible wasted object. move this to a private method getKinesisDataFetcherProviderConfig() and call this.dataFetcherProvider.apply(getKinesisDataFetcherProviderConfig());

Reply:
Not sure if it will ever be wasted in this context. The next line uses kinesisDataFetcherProviderConfig in both parts of the ternary statement